File: ActionsModule/ActionsLib/Team/Team.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Text.Json;
namespace ActionsLib
{
    public class Team
    {
        List<Player> _players;
        string _name;
        string _description;
        Color _mainColor;
        Color _liberoColor;
        public Team(string name, string description, Color mainColor, Color liberColor, List<Player> players = null)
        {
            if (players == null) _players = new List<Player>();
            else _players = new List<Player>(players);
            _name = name;
            _description = description;
            _mainColor = mainColor;
            _liberoColor = liberColor;
        }
        public Team(string name = "unknown", string description = "unknown", List<Player> players = null)
        {
            if (players == null) _players = new List<Player>();
            else _players = new List<Player>(players);
            _name = name;
            _description = description;
            _mainColor = Color.DarkGreen;
            _liberoColor = Color.Orange;
        }

        public string Name
        {
            get { return _name; }
        }
        public List<Player> Players
        {
            get { return _players; }
        }
        public string Description
        {
            get { return _description; }
        }
        public Color MainColor
        {
            get { return _mainColor; }
        }
        public Color LiberoColor
        {
            get { return _liberoColor; }
        }

        public void AddPlayer(Player player)
        {
            if (!_players.Contains(player)) _players.Add(player);
            else throw new Exception($"Team {_name} also contains Player {player.ToTableString()}");
        }
        public Player GetPlayerByNumber(int number)
        {
            foreach (Player player in _players)
            {
                if(player.Number == number) return player;
            }
            return null;
        }
        public void Save(StreamWriter sw)
        {
            sw.WriteLine(_name);
            sw.WriteLine(_description);
            sw.WriteLine(SaveColor(MainColor));
            sw.WriteLine(SaveColor(LiberoColor));
            sw.WriteLine(JsonSerializer.Serialize(Players));

        }
        private string SaveColor(Color c)
        {
            string res = $"{c.R},{c.G},{c.B}";
            return res;
        }
        private static Color LoadColor(string color)
        {
            string[] strs = color.Split(',');
            int[] values = new int[strs.Length];
            for (int i = 0; i < values.Length; i++) { values[i] = Convert.ToInt32(strs[i]); }
            return Color.FromArgb(values[0], values[1], values[2]);
        }
        public static Team Load(StreamReader sr)
        {
            string name = sr.ReadLine();
            string description = sr.ReadLine();
            Color color = LoadColor(sr.ReadLine());
            Color lcolor = LoadColor(sr.ReadLine());
            List<Player> players = JsonSerializer.Deserialize<List<Player>>(sr.ReadLine());

            return new Team(name, description, color, lcolor, players);
        }
    }
}

File: volleyball_stats/app/models/team.py
"""
Модель команды
"""
from typing import List, Optional, Tuple
from pydantic import BaseModel, Field
from .player import Player


class Team(BaseModel):
    """Модель волейбольной команды"""
    id: Optional[str] = Field(default=None, description="ID команды")
    name: str = Field(..., description="Название команды")
    description: str = Field(default="", description="Описание команды")
    main_color: Tuple[int, int, int] = Field(default=(0, 128, 0), description="Основной цвет команды (RGB)")
    libero_color: Tuple[int, int, int] = Field(default=(255, 165, 0), description="Цвет либеро (RGB)")
    players: List[Player] = Field(default_factory=list, description="Список игроков команды")

    class Config:
        json_schema_extra = {
            "example": {
                "name": "Зенит Казань",
                "description": "Профессиональная волейбольная команда",
                "main_color": [0, 0, 255],
                "libero_color": [255, 0, 0],
                "players": []
            }
        }

    def add_player(self, player: Player) -> None:
        """Добавить игрока в команду"""
        if player in self.players:
            raise ValueError(f"Команда {self.name} уже содержит игрока {player.to_table_string()}")
        self.players.append(player)

    def get_player_by_number(self, number: int) -> Optional[Player]:
        """Получить игрока по номеру"""
        for player in self.players:
            if player.number == number:
                return player
        return None

    def remove_player(self, number: int) -> bool:
        """Удалить игрока по номеру"""
        player = self.get_player_by_number(number)
        if player:
            self.players.remove(player)
            return True
        return False

    def get_players_by_amplua(self, amplua: int) -> List[Player]:
        """Получить игроков по амплуа"""
        return [p for p in self.players if p.amplua == amplua]
